samples/bluetooth/central_otc/src/main.c

```diff
@@ -0,0 +1,585 @@
+static struct gpio_callback button_cb_data;
+struct otc_btn_work_info {
+	struct k_work_delayable work;
+	uint32_t pins;
+} otc_btn_work;
+
+static void otc_btn_work_fn(struct k_work *work)
+{
+	struct k_work_delayable *dwork = k_work_delayable_from_work(work);
+	struct otc_btn_work_info *btn_work = CONTAINER_OF(dwork, struct otc_btn_work_info, work);
+	int err;
+	size_t size_to_write;
+
+	if (btn_work->pins == BIT(button.pin)) {
+		if (!first_selected) {
+			err = bt_ots_client_select_id(&otc, default_conn, BT_OTS_OBJ_ID_MIN);
+			first_selected = true;
+		} else {
+			printk("select next\n");
+			err = bt_ots_client_select_next(&otc, default_conn);
+		}
+		if (err) {
+			printk("Failed to select object\n");
+			return;
+		}
+		printk("Selecting object succeeded\n");
+	} else if (btn_work->pins == BIT(button1.pin)) {
+		printk("read OTS object meta\n");
+		err = bt_ots_client_read_object_metadata(&otc, default_conn,
+							 BT_OTS_METADATA_REQ_ALL);
+		if (err) {
+			printk("Failed to read object metadata\n");
+			return;
+		}
+	} else if (btn_work->pins == BIT(button2.pin)) {
+		size_to_write =
+			MIN(OBJ_MAX_SIZE, (otc.cur_object.size.alloc - otc.cur_object.size.cur));
+		memset(obj_data_buf, 0, size_to_write);
+		printk("write OTS object len %d\n", size_to_write);
+		for (uint32_t idx = 0; idx < size_to_write; idx++) {
+			obj_data_buf[idx] = UINT8_MAX - (idx % UINT8_MAX);
+		}
+		err = bt_ots_client_write_object_data(&otc, default_conn, obj_data_buf,
+						      size_to_write, 0,
+						      BT_OTS_OACP_WRITE_OP_MODE_NO_TRUNCATE);
+		if (err) {
+			printk("Failed to write object\n");
+			return;
+		}
+	} else if (btn_work->pins == BIT(button3.pin)) {
+		printk("read OTS object\n");
+		err = bt_ots_client_read_object_data(&otc, default_conn);
+		if (err) {
+			printk("Failed to read object %d\n", err);
+			return;
+		}
+	}
+}
+
+static void button_pressed(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	otc_btn_work.pins = pins;
+	k_work_schedule(&otc_btn_work.work, K_MSEC(100));
+}
+
+static void configure_button_irq(const struct gpio_dt_spec btn)
+{
+	int ret;
+
+	if (!device_is_ready(btn.port)) {
+		printk("Error: button device %s is not ready\n", btn.port->name);
+		return;
+	}
+
+	ret = gpio_pin_configure_dt(&btn, GPIO_INPUT);
+	if (ret != 0) {
+		printk("Error %d: failed to configure %s pin %d\n", ret, btn.port->name, btn.pin);
+		return;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&btn, GPIO_INT_EDGE_TO_ACTIVE);
+
+	if (ret != 0) {
+		printk("Error %d: failed to configure interrupt on %s pin %d\n", ret,
+		       btn.port->name, btn.pin);
+		return;
+	}
+	button_cb_data.pin_mask |= BIT(btn.pin);
+	gpio_add_callback(btn.port, &button_cb_data);
+
+	printk("Set up button at %s pin %d\n", btn.port->name, btn.pin);
+}
+
+static void configure_buttons(void)
+{
+	gpio_init_callback(&button_cb_data, button_pressed, 0);
+
+	for (uint8_t idx = 0; idx < BTN_COUNT; idx++) {
```

Comment:
```suggestion
	for (int idx = 0; idx < BTN_COUNT; idx++) {
```

`BTN_COUNT` is defined as an `int`